org settings: Display deactivated-user section to all.

YJDave opened the pull request:
Fixes #12290 

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
![image](https://user-images.githubusercontent.com/25907420/57502952-ccd73080-730b-11e9-9c03-e881a41e85ac.png)

![image](https://user-images.githubusercontent.com/25907420/57503327-4d4a6100-730d-11e9-972a-95242856f535.png)



<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
@rishig can you look at this?  I'm wondering if we want to remove the strikethrough in this view or something if we're enabling it...

rishig commented:
yeah, I think removing the strikethrough makes sense. If we can find an icon or something to represent deactivated we can put that after their names, but not important if nothing comes to mind.

YJDave commented:
@rishig updated with ban icon on deactivated users
![image](https://user-images.githubusercontent.com/25907420/58008418-f95c2b00-7b09-11e9-843a-907071297791.png)

![image](https://user-images.githubusercontent.com/25907420/58008489-20b2f800-7b0a-11e9-8367-293e08a1636c.png)

YJDave commented:
@rishig updated with ban icon on deactivated users
![image](https://user-images.githubusercontent.com/25907420/58008418-f95c2b00-7b09-11e9-843a-907071297791.png)

![image](https://user-images.githubusercontent.com/25907420/58008489-20b2f800-7b0a-11e9-8367-293e08a1636c.png)

rishig commented:
cool, thanks! I haven't looked at the CSS in detail, but it would be great to do something like
```
    color: inherit;
    margin-left: 2px;
    font-size: .9em;
```

With these it ends up looking like
![image](https://user-images.githubusercontent.com/890911/58074105-34397f80-7b59-11e9-8d77-29c8a764f2e7.png)

YJDave commented:
@rishig Thanks for the review. Updated style:
![image](https://user-images.githubusercontent.com/25907420/58083540-d21f5f80-7bd6-11e9-953b-6fca4bde397f.png)

rishig commented:
lgtm!

timabbott commented:
@YJDave can you take care of the rebase?

YJDave commented:
Updated!

timabbott commented:
This is great, but I think it may still be a bit too subtle whether the user is deactivated, especially if you just clicked "reactivate".  What do you think about changing the "Role" column to say e.g. "Deactivated <role>"?    I think that might provide some valuable extra reinforcement.

rishig commented:
yeah, that's a great idea. Maybe `Deactivated (was Member)`?

timabbott commented:
We can try that; Though it might be awkwardly long (keep in mind the screenshot above is missing an "actions" column since it's for a non-admin).

rishig commented:
ok, we can shorten to `Deactivated (Member)`, and if it ends up too long we can try something else.

YJDave commented:
@rishig @timabbott updated. Here is the new version:
It is only when admin deactivate/activate user, if you reload, it will automatically set to default user-role column. It only replaces strikethrough functionality.
![deactivateuser1](https://user-images.githubusercontent.com/25907420/59212190-60cc3e80-8bcf-11e9-9e4f-533c07f36d01.gif)

I have also tried another solution, as there are two tabs "Deactivated and Users". Here is the second,(which is not pushed):
![deactivateduser2](https://user-images.githubusercontent.com/25907420/59212825-02a05b00-8bd1-11e9-9828-66733324cb71.gif)

timabbott commented on static/js/settings_users.js:
```diff
@@ -28,7 +32,10 @@ function update_view_on_deactivate(row) {
+    var user_role = row.find(".user_role");
     row.find("button.open-user-form").show();
+    row.find('i.deactivated-user-icon').hide();
+    user_role.text(user_role.text().replace(/Deactivated|\(|\)/g, ""));
```
This parsing logic is messy, since the correctness depends on us always alternating calls to these two functions.

It's also broken with i18n where "Deactivated" will be a different word here; I think it might be better to either recompute or have a `data-orig-role` value that we can just pull out here.  

We could compute more cleanly above, too).

YJDave commented:
@timabbott yes, removed the logic and updated. 
Bots:
![image](https://user-images.githubusercontent.com/25907420/60795851-11573f00-a18a-11e9-9aa0-137988c3d8fa.png)
User list:
![image](https://user-images.githubusercontent.com/25907420/60795888-1d430100-a18a-11e9-8fd4-de3e3710d9e6.png)

Deactivated user:
![image](https://user-images.githubusercontent.com/25907420/60796796-bd4d5a00-a18b-11e9-9e97-6a0298756e88.png)

Before `wrapped-table` class:
![image](https://user-images.githubusercontent.com/25907420/60797006-20d78780-a18c-11e9-8e42-8436426f9549.png)

@pragatiagrawal31 is working on table filter/search, I think adding "Deactivated/Reactivated" filter in both view will help.

timabbott commented:
Great, merged, thanks @YJDave!  

@rishig FYI in case there's anything else we want to do with the users/bots/deactivated users pages as follow-ups.  The main thing I'm aware of that's in flight is @pragatiagrawal31's work on making all the columns sortable, which should merge cleanly with this.
